File: src/main/java/aginsun/kingdoms/server/entities/EntityFisher.java
package aginsun.kingdoms.server.entities;

import aginsun.kingdoms.api.entities.EntityNPC;
import aginsun.kingdoms.server.TaleOfKingdoms;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.init.Items;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;

import static aginsun.kingdoms.server.handlers.GuiHandler.GUI_FISHER;

public final class EntityFisher extends EntityNPC
{
    public EntityFisher(World world)
    {
        super(world, new ItemStack(Items.fishing_rod), 40.0F);
    }

    @Override
    public boolean canBePushed()
    {
        return false;
    }

    @Override
    protected boolean isMovementCeased()
    {
        return true;
    }

    @Override
    public boolean interact(EntityPlayer player)
    {
        if (this.canInteractWith(player))
        {
            player.openGui(TaleOfKingdoms.instance, GUI_FISHER, worldObj, 0, 0, 0);
        }
        return true;
    }
}

File: src/main/java/kingdoms/server/entities/EntityFisher.kt
package kingdoms.server.entities

import kingdoms.api.entities.EntityNPC
import kingdoms.server.handlers.NetworkHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.init.Items
import net.minecraft.item.ItemStack
import net.minecraft.world.World

class EntityFisher(world: World): EntityNPC(world, ItemStack(Items.fishing_rod), 40.0F)
{
    override fun canBePushed(): Boolean
    {
        return false
    }

    override fun isMovementCeased(): Boolean
    {
        return true
    }

    override fun interact(player: EntityPlayer): Boolean
    {
        if (!worldObj.isRemote)
        {
            if (this.canInteractWith(player))
            {
                NetworkHandler.openGui(this.entityId, player)
            }
        }
        return true
    }
}